### PR TITLE
Improve appimagelauncherd

### PR DESCRIPTION
**Pull request by azubieta**
Small set of improvements focused on simplifying the code base and make a proper use of the Qt threading.

### CODE REVIEW

File: src/daemon/worker.cpp [hunk 1]
```diff
@@ -51,36 +47,25 @@ class Worker::PrivateData {
             const auto isAppImage = 0 < appImageType && appImageType <= 2;
 
             if (type == INTEGRATE) {
-                mutex->lock();
```
Comment (TheAssassin) [inline src/daemon/worker.cpp]: Seems like none of the outputs is now synchronized any more. Doesn't that garble the output? That's the only reason I introduced this in the first place.

File: src/daemon/worker.cpp [hunk 2]
```diff
@@ -144,47 +129,28 @@ void Worker::executeDeferredOperations() {
...
-    }
-
-    // start timer and notify future calls to this that a timer is already running
-    d->timerActive = true;
-
-    QTimer::singleShot(d->TIMEOUT, [this]() {
-        d->timerActive = false;
-        executeDeferredOperations();
-    });
+    if (!d->deferredOperationsTimer.isActive())
```
Comment (TheAssassin) [inline src/daemon/worker.cpp]: Is that line really still needed? You `connect`ed signals above in line 105f. already.

File: src/fswatcher/filesystemwatcher.cpp [hunk 3]
```diff
@@ -150,24 +157,17 @@ bool FileSystemWatcher::stopWatching() {
...
-                emit fileCreated(event.path);
-            } else if (mask & d->fileDeletionEvents) {
-                emit fileDeleted(event.path);
-            } else if (mask & d->fileModificationEvents) {
-                emit fileModified(event.path);
-            }
+void FileSystemWatcher::readEvents() {
+    auto events = d->readEventsFromFd();
+
+    for (const auto& event : events) {
```
Comment (TheAssassin) [inline src/fswatcher/filesystemwatcher.cpp]: Seems a little more compact indeed.

File: src/fswatcher/filesystemwatcher.cpp [hunk 4]
```diff
@@ -101,6 +103,7 @@ class FileSystemWatcher::PrivateData {
             }
 
             watchFdMap[watchFd] = directory;
+            eventsLoopTimer.start();
```
Comment (TheAssassin) [inline src/fswatcher/filesystemwatcher.cpp]: How to stop a fs watcher now? Shouldn't we introduce a `stop()` method now that it's possible? Do you have a plan for that already, e.g., in some other PR?

File: src/daemon/worker.cpp (hunk 2 above)
Comment (azubieta) [reply src/daemon/worker.cpp]: Every time we call `start` on the timer it gets restarted. So we still need to check that is not running before calling start.

File: src/daemon/worker.cpp (hunk 1 above)
Comment (azubieta) [reply src/daemon/worker.cpp]: Haven't considered that. Let me re-check it.

File: src/fswatcher/filesystemwatcher.cpp (hunk 4 above)
Comment (azubieta) [reply src/fswatcher/filesystemwatcher.cpp]: There is a stop method on the `FSWatcher`
https://github.com/TheAssassin/AppImageLauncher/blob/90de238f5a952e619791b9aad4a000a45fe6462b/src/fswatcher/filesystemwatcher.h#L30

Comment (azubieta) [reply src/fswatcher/filesystemwatcher.cpp]: Wonder why I used a `QThread` class there.

Comment (azubieta) [reply src/fswatcher/filesystemwatcher.cpp]: fixed

File: src/daemon/worker.cpp (hunk 2 above)
Comment (TheAssassin) [reply src/daemon/worker.cpp]: Fair enough.

File: src/daemon/worker.cpp (hunk 1 above)
Comment (azubieta) [reply src/daemon/worker.cpp]: Feature restored.

**Pull request merged**